meetings/2022-05-11.md

```diff
@@ -0,0 +1,102 @@
+* ...: In OAuth access tokens are opaque,
+* LD: I like that idea of having access modes as common interface
+* JB: I agree with that too.
+* EP: Access tokens are defined as opaque to the client only.
+* ...: Also the AS might need to know a bit about the data to evaluate a request. For example for inherited access.
+* JB: Could it be if you used some access control language, that either at the AS or another level.??? you could use a Data Grant or Access Grant and transform that into a ticket. You could model that using ACP. If the AS is using ACP to process and turn something into modes... Laurens says he goes straight from Data Grants to modes. But in that case, you would need something to calculate that relation.
+* Matthieu: I think that ultimatelly the problem is what is the authoritative source for authorization. You wnat to be able to enforce set of contitions, for example by ACP. But than you open authorization to a paralel system. You could end up with a situation that conditions you try to enforce on acp level is not being enforced by other system. For example in ACP you can grant access based on Verifiable Credential. One could have VC based way of verifying credentials. On ACP side it is only require to check if VC access is enabled for the resource.
+* ...: If we allow Data Grant based access to resources. We would need to say that access to given resource would be for example at most read. It would require to agree who is the ultimate source of the authorization.
+* ...: There is also a case if data grants grants you an access over multiple resources.
+* EP: I just have a quick comment that while talking about Access Modes added in access token, we don't want to get an access token for every request. We shouldn't require the client to get a new token for every request. I don't think the access token should be specific for each request.
+* MB: Yes, as far as I understand, the ID token defined in Solid OIDC, you only claim the identity of requesting agent and issuer/client... But the authorization evaluation needs to happen on every request.
+* EP: The request would include access token, so the authz server can correlate what it knows about the user.
+* LD: I think it's quite hard to prevent token re-issuance. I'll get back to my implementation [on GitHub](https://github.com/laurensdeb/interoperability). Right now, you need token reissuance with every request.
+* ...: When you get a 401 with a token previously got, you can add claims to your token and get new tokens with lesser overhead but it's quite complex to avoid reissuance. You get a ticket with the access modes requested. The access tokens pass information between AS and RS and also give to the token a JTI (token id). You could speed up the work of the UMA AS.
```

Comment:
```suggestion
* ...: When you get a 401 with a token previously used, you can add claims to your token and get new tokens with less overhead, but it's quite complex to avoid reissuance. You get a ticket with the access modes requested. The access tokens pass information between AS and RS and also give to the token a JTI (token id). You could speed up the work of the UMA AS.
```